nsls2forge_utils/all_feedstocks.py

```diff
@@ -0,0 +1,109 @@
+    (e.g. nsls-ii-forge).
+
+    Parameters
+    ----------
+    organization: str
+        Name of organization on GitHub.
+    username: str, optional
+        Name of user on GitHub for authentication.
+        Uses environment variables if not specified.
+    password: str, optional
+        Password of user on GitHub for authentication.
+        Uses environment variables if not specified.
+
+    Returns
+    -------
+    names: list
+        List of repository names that end with '-feedstock' (stripped).
+    '''
+    if username == '':
+        gh = github3.login(os.environ['USERNAME'], os.environ['PASSWORD'])
+    else:
+        gh = github3.login(username, password)
+    org = gh.organization(organization)
+    repos = org.repositories()
+    names = []
+    try:
+        for repo in repos:
+            name = repo.name
+            if name.endswith("-feedstock"):
+                name = name.split("-feedstock")[0]
+                logger.info(name)
+                names.append(name)
+    except github3.GitHubError:
+        msg = ["Github rate limited. "]
+        c = gh.rate_limit()["resources"]["core"]
+        if c["remaining"] == 0:
+            ts = c["reset"]
+            msg.append("API timeout, API returns at")
+            msg.append(
+                datetime.datetime.utcfromtimestamp(ts).strftime("%Y-%m-%dT%H:%M:%SZ"),
+            )
+        logger.warning(" ".join(msg))
+        raise
+    return names
+
+
+def get_all_feedstocks(organization, username='', password='', cached=False):
+    '''
+    Gets all feedstocks either from GitHub or from names.txt if flag is specified.
+
+    Parameters
+    ----------
+    organization: str
+        Name of organization on GitHub.
+    username: str, optional
+        Name of user on GitHub for authentication.
+        Uses environment variable if not specified.
+    password: str, optional
+        Password of user on GitHub for authentication.
+        Uses environment variable if username not specified.
+    cached: bool, optional
+        Specified if client wants to take repository names from names.txt.
+
+    Returns
+    -------
+    names: list or None
+        List of repository names that end with '-feedstock' (stripped).
+        None if no organization or username is specified.
+    '''
+    if cached:
+        logger.info("reading names")
+        with open("names.txt", "r") as f:
+            names = f.read().split()
+        return names
+    if organization == '':
+        logger.info("No GitHub organization specified.")
+        return None
+    names = get_all_feedstocks_from_github(organization, username, password)
+    return names
+
+
+def main(args=None):
+    organization = 'nsls-ii-forge'
+    names = get_all_feedstocks(organization, cached=False)
+    # write each repository name to a file
+    with open("names.txt", "w") as f:
+        for name in names:
+            f.write(name)
+            f.write("\n")
```

Comment:
We can reduce a number of lines here with updating the previous line with:
```py
f.write('{}\n'.format(name))  # python<3.6 compatible
```
or
```py
f.write(f'{name}\n')  # python>=3.6 f-string feature
```